networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/provider.py

```diff
@@ -0,0 +1,130 @@
+class Provider(object):
+    """
+    Provider interface used for realization of OpenStack objects 
```

Comment:
If this is supposed to be an interface, would it make sense to enforce it via [Abstract Base Class](https://docs.python.org/3/library/abc.html)?